src/components/admin/form-components/withSearchBar.jsx

```diff
@@ -1,223 +1,94 @@
+      }, debounceTime);
     }
 
-    componentDidUpdate() {
-      this.debouncedGetSuggestions();
-    }
-
-    componentWillReceiveProps() {
-      this.safeSetState({
-        searchValue: '',
-        searchSuggestions: [],
-      });
-    }
-
-    handleClick(article) {
-      this.safeSetState({
-        searchValue: '',
-        searchSuggestions: [],
-      });
-      this.props.handleClick(article);
+    handleChange(query) {
+      this.debounceSuggestions(query);
     }
 
     render() {
-      const searchBarClass = 'search-bar';
-      const searchBarResultClass = 'search-bar-result';
-      const slug = this.props.enableAdd
-        ? slugify(config.category, this.state.searchValue)
-        : '';
       return (
-        <div
-          className={`${searchBarClass} ${searchBarClass}-${config.category}`}
-        >
-          <TextField
-            floatingLabelText={`Search for ${capFirstLetter(config.category)}`}
-            hintText={capFirstLetter(config.category)}
-            fullWidth={this.props.fullWidth}
-            value={this.state.searchValue}
-            onChange={this.fieldUpdaters.searchValue}
-          />
-          <div>
-            <Menu style={!this.props.fullWidth && { width: 200 }}>
-              {this.props.enableAdd &&
-                this.state.searchValue &&
-                this.state.searchSuggestions.length === 0 && (
-                  <div
-                    className={`${searchBarResultClass} search-bar-${
-                      config.category
-                    }`}
-                    key={slug}
-                  >
-                    {/* eslint-disable react/jsx-no-bind */}
-                    <MenuItem
-                      leftIcon={<ContentAdd />}
-                      primaryText={this.state.searchValue}
-                      onClick={this.handleClick.bind(this, {
-                        value: this.state.searchValue,
-                        slug,
-                        id: null,
-                      })}
-                      disabled={this.props.disabled}
-                    />
-                    {/* eslint-enable react/jsx-no-bind */}
-                  </div>
-                )}
-              {/* eslint-disable react/jsx-no-bind */
-              this.state.searchSuggestions.map(item => (
-                <div
-                  className={`search-bar-result search-bar-${config.category}`}
-                  key={item.slug}
-                >
-                  <MenuItem
-                    primaryText={config.formatter(item)}
-                    onClick={this.handleClick.bind(this, item)}
-                    disabled={this.props.disabled}
-                  />
-                </div>
-              ))
-              /* eslint-enable react/jsx-no-bind */
-              }
-            </Menu>
-          </div>
-        </div>
+        <WrappedField
+          suggestions={this.state.suggestions}
+          updateSuggestions={this.handleChange}
```

Comment:
I guess I am a little bit confused by this HOC name. From what it looks like `withSearchableConcept` is wrapping `SearchBar` in Falcor logic correct? I think, `withFalcorSearch` would be a clearer name for that idea.

Reply:
@frederikbrinck and that would allow you to remove the suggestions prop here.

Reply:
@frederikbrinck I like `withFalcorSearch` too. (@Helw150 I guess all these show as outdated because you accidentally were looking at the not newest commit while reviewing?)

Reply:
Got it!